File: py/leetcode-uniquenumberofoccurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        m = {}
        n = {}

        for i in arr:
            if i not in m.keys():
                m[i] = 1
            else:
                m[i] += 1
        
        for i, k in m.items():
            if k not in n.keys():
                n[k] = 1
            else:
                n[k] += 1
        
        return not any(x != 1 for x in n.values())

File: cpp/leetcode-uniquenumberofoccurrences.cpp
#include <unordered_map>
#include <vector>
using namespace std;
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> m;

        for (int &k : arr) {
            m[k] += 1;
        }

        unordered_map<int, int> o;
        for (const auto& n : m) {
            o[n.second] += 1;
        }

        for (const auto& n : o) {
            if (n.second != 1) return false;
        }

        return true;

    }
};